src/vorta/views/partials/password_input.py

```diff
@@ -0,0 +1,127 @@
+from PyQt5 import QtCore
+from PyQt5.QtWidgets import QAction, QFormLayout, QLabel, QLineEdit, QWidget
+from vorta.i18n import trans_late, translate
+from vorta.views.utils import get_colored_icon
+
+
+class PasswordLineEdit(QLineEdit):
+    password_too_short = QtCore.pyqtSignal()
+
+    def __init__(self, parent=None, show_visibility_button=True):
+        super().__init__(parent)
+
+        self._show_visibility_button = show_visibility_button
+
+        self.setEchoMode(QLineEdit.Password)
+
+        if self._show_visibility_button:
+            self.showHideAction = QAction(self.tr("Show password"), self)
+            self.showHideAction.setCheckable(True)
+            self.showHideAction.toggled.connect(self._toggle_password_visibility)
+            self.showHideAction.setIcon(get_colored_icon("eye"))
+            self.addAction(self.showHideAction, QLineEdit.TrailingPosition)
+
+    def get_password(self):
+        """Return password text"""
+        return self.text()
+
+    def _toggle_password_visibility(self):
+        if self.echoMode() == QLineEdit.Password:
+            self.setEchoMode(QLineEdit.Normal)
+            self.showHideAction.setIcon(get_colored_icon("eye"))
```

Comment:
Since it will never be connected if `_show_visibility_button` is false, it shouldn't raise an error

https://github.com/borgbase/vorta/blob/ac7de01645b0ab064fd80eb2543092472a5adee9/src/vorta/views/partials/password_input.py#L17-L20

Do you want to make `show_visibility_button` or `_toggle_password_visibility` public?

Reply:
Ignore the above comment. I somehow missed the review above.

Reply:
This is going in the right direction. However you made a logical error. This still fails if `_show_visibility_button` is false.